components/Map.js

```diff
@@ -1,30 +1,19 @@
-import {
-  GoogleMap,
-  Marker,
-  useLoadScript,
-  InfoWindow,
-} from "@react-google-maps/api";
-import {useEffect, useState, React, useMemo, Fragment} from "react";
+import {GoogleMap, Marker, InfoWindow} from "@react-google-maps/api";
+import {useEffect, React, useMemo, useContext} from "react";
+import AppContext from "./AppContext";
 import styled from "styled-components";
 const containerStyle = {
   width: "75vw",
   height: "50vh",
   borderRadius: "10px",
   border: "1px solid #572887",
 };
-const libraries = ["places", "geometry"];
 
 export default function Map() {
-  const [status, setStatus] = useState(null);
-  const [center, setCenter] = useState(null);
-  const [clickedMarker, setClickedMarker] = useState(null);
-  const [lawyers, setLawyers] = useState([]);
-  const {isLoaded} = useLoadScript({
-    googleMapsApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY,
-    libraries,
-  });
+  const context = useContext(AppContext);
+
   const onLoadMap = map => {
-    loadLawyers(center, map);
+    loadLawyers(context.center, map);
```

Comment:
This is a handler so you should call it `handleLoadMap`.  We want to prefix the event props with `on...` and the corresponding handlers with `handle...`:

```js
onLoad={handleLoadMap}
```